### DISCUSSION TITLE
Rename Column Header without using expressions?

### DISCUSSION THREAD

**Discussion by nekw** [perspective-dev/perspective#1507] [Q&A] [answered]
In Perspective Viewer, I want to rename a column header name e.g. from 'market-segment' to 'Market Segment'.
Knowing that we can use expressions to create column alias name. But, we will have an extra column which is redundant and confusing users. Is there a way to achieve it without having extra column? Thanks!

**Comment by texodus** [comment, marked as answer]
No, and this is not possible to implement the way you describe without breaking Perspective's persistence guarantee, namely that calling `save()` is guaranteed to work with `restore()` as long as the `Table()` schema under which it was constructed remains in-tact.  Mutating the structure of the underlying `Table()` in-place will fail to resolve renamed columns unless we also persist the alias map along with it, which is exactly what a simple expression column with just an alias does (in addition to supplanting column duplicates, multi-aggregate columns, and a host of other minor features). 

This may be "confusing" in some sense, but it represents the actual state of the viewer and is important information for the user to reason about what will occur with their Perspective visualization when they `save()`/`restore()`/`reset()`/`reload()` or otherwise mutate a viewer's state.

All this being said, it sounds like what you _actually_ want is to rename the column in advance, as the developer of an application with an embedded `<perspective-viewer>`.  If this is the case, I recommend you just rename the column in the data source itself before you load it, and you won't need aliased columns at all;  Perspective does not have any character restrictions on column names explicitly to avoid this sort of artificial API complexity.
